src/system/version_migration/versions/v1_1_0.ts

```diff
@@ -0,0 +1,29 @@
+import { SessionSaveData } from "../../game-data";
+import { CustomPokemonData } from "#app/data/custom-pokemon-data";
+
+export const systemMigrators = [] as const;
+
+export const settingsMigrators = [] as const;
+
+export const sessionMigrators = [
+  /**
+   *  Converts old Pokemon natureOverride and mysteryEncounterData
+   *  to use the new conjoined {@linkcode Pokemon.customPokemonData} structure instead.
+   *  @param data {@linkcode SessionSaveData}
+   */
+  function migrateCustomPokemonDataAndNatureOverrides(data: SessionSaveData) {
+    // Fix Pokemon nature overrides and custom data migration
+    data.party.forEach(pokemon => {
+      if (pokemon["mysteryEncounterData"]) {
+        pokemon.customPokemonData = new CustomPokemonData(pokemon["mysteryEncounterData"]);
+      }
+      if (pokemon["fusionMysteryEncounterData"]) {
+        pokemon.fusionCustomPokemonData = new CustomPokemonData(pokemon["fusionMysteryEncounterData"]);
+      }
+      pokemon.customPokemonData = pokemon.customPokemonData ?? new CustomPokemonData();
+      if (pokemon["natureOverride"]) {
```

Comment:
```suggestion
      if (pokemon["natureOverride"] && pokemon["natureOverride"] !== -1) {
```
I'm not entirely sure this could cause issues, but better safe than sorry probably, we don't want to override a potentially existing nature override with -1